File: Problems_and_Solutions/day-7/Java/Solution2.java
class Solution2 {
    public static boolean check(int[] nums) {
        int n = nums.length;
        int count = 0;

        for(int i=0; i<n; i++){
            if(nums[i] > nums[(i+1) % n]){
                count++;
            }
        }
        return count <= 1;
    }
    
    public static void main(String[] args){
        int nums[] = {3, 4, 5, 1, 2};
        System.out.println(check(nums));
    }
}

File: Problems_and_Solutions/day-7/cpp/solution 2.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int increasingOrderFailed=0;
        int noOfComparisons=0;
        int n=nums.size();
        for(int i=0;noOfComparisons<n;i++)
        {
            if(nums[i]>nums[(i+1)%n])
            {
                increasingOrderFailed++;
                if(increasingOrderFailed>1)
                    return false;
            }
            noOfComparisons++;
        }
        return true;
    }
};